Sources/Common/DataModel/Box/index.js

```diff
@@ -0,0 +1,205 @@
+      }
+    } else {
+      coord[i] = candidatePlane[i];
+    }
+  }
+
+  return 1;
+}
+
+// ----------------------------------------------------------------------------
+// Static API
+// ----------------------------------------------------------------------------
+
+export const STATIC = {
+  intersectBox,
+};
+
+// ----------------------------------------------------------------------------
+// vtkBox methods
+// ----------------------------------------------------------------------------
+
+function vtkBox(publicAPI, model) {
+  // Set our className
+  model.classHierarchy.push('vtkBox');
+
+  publicAPI.setBounds = (...bounds) => {
+    let boundsArray = [];
+
+    if (Array.isArray(bounds[0])) {
+      boundsArray = bounds[0];
+    } else {
+      for (let i = 0; i < bounds.length; i++) {
+        boundsArray.push(bounds[i]);
+      }
+    }
+
+    if (boundsArray.length !== 6) {
+      return;
+    }
+
+    model.bbox.setBounds(boundsArray);
+  };
+
+  publicAPI.getBounds = () => model.bbox.getBounds();
+
+  publicAPI.evaluateFunction = (...x) => {
+    let point = [].concat(x);
+    if (Array.isArray(x[0])) {
+      point = x[0];
+    }
+
+    if (point.length !== 3) {
+      return Number.MAX_VALUE;
+    }
+
+    let diff;
+    let dist;
+    let t;
+    let minDistance = -Number.MAX_VALUE;
+    let distance = 0;
+    const minPoint = model.bbox.getMinPoint();
+    const maxPoint = model.bbox.getMaxPoint();
+    let inside = 1;
+    for (let i = 0; i < 3; i++) {
+      diff = model.bbox.getLength(i);
+      if (diff !== 0.0) {
+        t = (point[i] - minPoint[i]) / diff;
+        if (t < 0.0) {
+          inside = 0;
+          dist = minPoint[i] - point[i];
+        } else if (t > 1.0) {
+          inside = 0;
+          dist = point[i] - maxPoint[i];
+        } else { // want negative distance, we are inside
+          if (t <= 0.5) {
+            dist = minPoint[i] - point[i];
+          } else {
+            dist = point[i] - maxPoint[i];
+          }
+          if (dist > minDistance) { // remember, it's engative
+            minDistance = dist;
+          }
+        } // end if inside
+      } else {
+        dist = Math.abs(point[i] - minPoint[i]);
+        if (dist > 0.0) {
+          inside = 0;
+        }
+      }
+      if (dist > 0.0) {
+        distance += dist * dist;
+      }
+    } // end for i
+    distance = Math.sqrt(distance);
+    if (inside) {
+      return minDistance;
+    }
+    return distance;
+  };
+}
+
+// ----------------------------------------------------------------------------
+// Object factory
+// ----------------------------------------------------------------------------
+
+const DEFAULT_VALUES = {
+};
+
+// ----------------------------------------------------------------------------
+
+export function extend(publicAPI, model, initialValues = {}) {
+  Object.assign(model, DEFAULT_VALUES, initialValues);
+
+  // Object methods
+  macro.obj(publicAPI, model);
+
+  if (!model.bbox) {
```

Comment:
We could improve that also support when we are getting a serialized version of the bbox. But I could not find a good example where I've already done it.

So it could be good for now. We can always improve that later when we bump into it...